pandas/core/indexes/base.py

```diff
@@ -4717,12 +4722,13 @@ def argsort(self, *args, **kwargs) -> np.ndarray:
         >>> idx[order]
         Index(['a', 'b', 'c', 'd'], dtype='object')
         """
-        result = self.asi8
-
-        if result is None:
-            result = np.array(self)
+        if needs_i8_conversion(self.dtype):
+            # TODO: these do not match the underlying EA argsort methods GH#37863
+            return self.asi8.argsort(*args, **kwargs)
```

Comment:
why are you calling .asi8 here?

Reply:
comment just above #37863 a handful of tests fail if we pass through to the backing EA.argsort

Reply:
won't these show the deprecation warning?

Reply:
no, because we're not deprecating it for the datetimelike classes